### PR TITLE
docs: Apply feedback from 24.09 VDR

### PR DESCRIPTION
**Pull request by terrykong**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by ashors1 — APPROVED** [review]
Thanks for the PR!

**Comment by zredeaux07** [comment]
Reviewed the updates and made a few minor grammatical edits.

**Review by ashors1 — APPROVED** [review]

File: docs/user-guide/cai.rst [hunk 1]
```diff
@@ -14,7 +16,7 @@ CAI allows training a harmless, but non-evasive AI assistant that engages with h
 .. _Constitutional AI (CAI): https://arxiv.org/abs/2212.08073
 
 CAI
-###############
+###
 The basic steps of CAI are described in this section and illustrated in the figure below (`Figure 1 <https://arxiv.org/abs/2212.08073>`_).
```
Comment (jgerh) [inline docs/user-guide/cai.rst:20]: Check link <https://arxiv.org/abs/2212.08073>. This link points to an arkiv article, Constitutional AI: Harmlessness from AI Feedback, not the "figure below" in this file.

File: docs/user-guide/cai.rst [hunk 2]
```diff
@@ -41,7 +43,7 @@ Simplicity and transparency: encoding the training goals in a simple list of nat
 Reducing iteration time: obviating the need to collect new human feedback labels when altering the objective or testing different behaviors.
 
 Train a CAI model
```
Comment (jgerh) [inline docs/user-guide/cai.rst:45]: fix capitalization

Train a CAI Model

File: docs/user-guide/cai.rst [hunk 3]
```diff
@@ -62,15 +64,10 @@ This section is a step-by-step tutorial that walks you through how to run a full
...
-   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
-
-   If you run into any problems, refer to NeMo's `Known Issues page <https://docs.nvidia.com/nemo-framework/user-guide/latest/knownissues.html>`__. The page enumerates known issues and provides suggested workarounds where appropriate.
-
 .. image:: ../assets/cai_flow.png
 
 Step 1: Download models and datasets
-#############################################################################
+####################################
 1. Download ``Mistral-7B-Instruct`` and ``Mistral-7B`` LLM models from https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1 and https://huggingface.co/mistralai/Mistral-7B-v0.1 into the models folder.
```
Comment (jgerh) [inline docs/user-guide/cai.rst:71]: add definitive article "the"

1. Download the ``Mistral-7B-Instruct`` and ``Mistral-7B`` LLM models from https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1 and https://huggingface.co/mistralai/Mistral-7B-v0.1 into the models folder.

File: docs/user-guide/cai.rst [hunk 4]
```diff
@@ -62,15 +64,10 @@ This section is a step-by-step tutorial that walks you through how to run a full
...
 
    Then, convert into .nemo format:
```
Comment (jgerh) [inline docs/user-guide/cai.rst:73]: revise

Then, convert them into .nemo format:

File: docs/user-guide/cai.rst [hunk 5]
```diff
@@ -62,15 +64,10 @@ This section is a step-by-step tutorial that walks you through how to run a full
...
 7. Run inference.
 
-.. note::
-   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
-
-   If you run into any problems, refer to NeMo's `Known Issues page <https://docs.nvidia.com/nemo-framework/user-guide/latest/knownissues.html>`__. The page enumerates known issues and provides suggested workarounds where appropriate.
-
 .. image:: ../assets/cai_flow.png
 
 Step 1: Download models and datasets
```
Comment (jgerh) [inline docs/user-guide/cai.rst:69]: revise

Step 1: Download the models and datasets

File: docs/user-guide/cai.rst (hunk 1 above)
Comment (jgerh) [inline docs/user-guide/cai.rst:20]: please check the link. This link, (`Figure 1 <https://arxiv.org/abs/2212.08073>`_), goes to an arkiv article, not the Constitutitional AI Flow figure below.

File: docs/user-guide/cai.rst [hunk 6]
```diff
@@ -14,7 +16,7 @@ CAI allows training a harmless, but non-evasive AI assistant that engages with h
 .. _Constitutional AI (CAI): https://arxiv.org/abs/2212.08073
 
 CAI
-###############
+###
 The basic steps of CAI are described in this section and illustrated in the figure below (`Figure 1 <https://arxiv.org/abs/2212.08073>`_).
```
Comment (jgerh) [inline docs/user-guide/cai.rst:21]: Line 22/24

fix punctuation (fine-tuned)


(RL Stage) AI Comparison Evaluations → Reward Model → Reinforcement Learning: The AI generates pairs of responses to harmfulness prompts using the fine-tuned model, then evaluates which response is better according to a principle in the constitution, and then trains a reward model based on this dataset of AI preferences and a human helpfulness preferences. The AI then trains with RL using the learned reward model.

File: docs/user-guide/cai.rst [hunk 7]
```diff
@@ -29,7 +31,7 @@ The basic steps of CAI are described in this section and illustrated in the figu
 Critiques, revisions, and AI harmlessness feedback are steered by a small set of principles drawn from a ‘constitution’. The supervised stage significantly improves the initial model. It gives some control over the initial behavior at the start of the RL phase, while addressing potential exploration problems. The RL stage significantly improves performance and reliability.
 
 Motivation
-###############
+##########
 Constitutional AI motivation refers to designing AI systems in such a way that their objectives and behaviors are guided by a set of predefined rules or principles. It includes the following:
 
 Scaling supervision: using AI to help humans supervise other AIs more efficiently and effectively, especially for tasks where AI capabilities may exceed human ones.
```
Comment (jgerh) [inline docs/user-guide/cai.rst:37]: Line 35/37 through Line 41/43

change to bulleted list, fix capitalization, and revise text (suggestion)

- Scaling Supervision: Use AI to assist humans in supervising other AIs more efficiently and effectively, particularly for tasks where AI capabilities may surpass human ones.
- 
- A Harmless but Non-Evasive Assistant: Minimize the tension between helpfulness and harmlessness, and avoid evasive responses that reduce transparency and helpfulness.
- 
- Simplicity and Transparency: Encode training goals in a straightforward list of natural language instructions or principles, and employ chain-of-thought reasoning to make AI decision-making explicit and understandable.
- 
- Reducing Iteration Time: Eliminate the need to collect new human feedback labels when modifying objectives or testing different behaviors.
-

File: docs/user-guide/cai.rst [hunk 8]
```diff
@@ -112,7 +109,7 @@ Step 1: Download models and datasets
 
 
 Step 2: Generate and revise responses to harmful prompts creating the SL-CAI dataset
-###################################################################################################
+####################################################################################
```
Comment (jgerh) [inline docs/user-guide/cai.rst:112]: Line 161/158 fix punctuation

This will generate an SL-CAI dataset of prompts and revised responses as ``cai_revisions_aligner_chat_template.json``.

Line 163/160 revise

The few-shot samples should be provided following the template in ``few_shot_samples_example.json``. Fill in the `content` tags and choose how many samples to use. The samples should include a red teaming prompt, a response from the helpful model (e.g., ``Mistral-7B`` in this tutorial), critique and revision requests, and responses. An example is shown in the `Anthropic repo <https://github.com/anthropics/ConstitutionalHarmlessnessPaper/blob/main/prompts/CritiqueRevisionFewShotPrompts.json>`_.

Line 165/162 use .rst note syntax and revise

.. note::
    The tokenizer file can be found by extracting the .nemo checkpoint using `tar -xf /models/mistral/mistral-7b-Instruct.nemo`. There are two tokenizer files that end with `.model` in the model checkpoint, and they are identical. You can use either one for data processing.

File: docs/user-guide/cai.rst [hunk 9]
```diff
@@ -199,7 +196,7 @@ Note that you would need to set up multi-node training run in your cluster env,
 
 
 Step 4: Generate the RL-CAI (preference) dataset for RM and PPO training
-##############################################################################################################
+########################################################################
 
 The following section runs an inference server with the SL-CAI model that we've previously trained, and queries it with red teaming prompts asking for several responses per prompt.
 The responses will then be ranked by a judge LLM being run from NVIDIA's NGC. An NGC API key can be acquired `here`_.
```
Comment (jgerh) [inline docs/user-guide/cai.rst:202]: Line 204/201 to 205/202 revise

The following section runs an inference server with the SL-CAI model that we've previously trained. It queries the server with red teaming prompts, requesting several responses per prompt. These responses will then be ranked by a judge LLM running from NVIDIA's NGC. You can acquire an NGC API key `here`_.

File: docs/user-guide/cai.rst [hunk 10]
```diff
@@ -258,7 +255,7 @@ This command will create the ``rl-cai`` dataset files in the defined output fold
 
 
 Step 5: Train the RM
```
Comment (jgerh) [inline docs/user-guide/cai.rst:257]: Line 260/257 revise. define acronym which is useful for user searches

Step 5: Train the Reward Model (RM)

File: docs/user-guide/cai.rst [hunk 11]
```diff
@@ -330,7 +327,7 @@ Run the following command to launch actor training and a reference policy server
 The trained LLM policy checkpoint will be saved to the output dir given by ``exp_manager.explicit_log_dir``.
 
 Step 7: Inference
```
Comment (jgerh) [inline docs/user-guide/cai.rst:329]: Line 332/329 revise step text so it is consistent with above and fix capitalization

Step 7: Run inference

File: docs/user-guide/knowledge-distillation.rst [hunk 12]
```diff
@@ -1,5 +1,9 @@
 .. include:: /content/nemo.rsts
 
+.. include:: aligner-algo-header.rst
+
+.. _nemo-aligner-knowledge-distillation:
+
```
Comment (jgerh) [inline docs/user-guide/knowledge-distillation.rst:6]: Line 18/22 use capitalization in title

Obtain the Fine-Tuned Teacher and Pre-Trained Student Models

Line 20/24 fix punctuation

To start, we must first download both the pre-trained student and fine-tuned teacher models.

Line 45/49 fix punctuation

After these steps, you should have files ``2b_student.nemo`` and ``teacher_checkpoint/Nemotron-3-8B-Chat-4k-SFT.nemo`` to use in NeMo-Aligner.

File: docs/user-guide/nemoaligner.rsts [hunk 13]
```diff
@@ -1,8 +1,8 @@
...
-.. _model-aligner-intro:
+.. _nemo-aligner-intro:
 
-Model Alignment
-!!!!!!!!!!!!!!!
+NeMo-Aligner
+!!!!!!!!!!!!
 
 Introduction
 ############
```
Comment (jgerh) [inline docs/user-guide/nemoaligner.rsts:8]: Delete subheading "Introduction". We usually don't stack titles and headings without intervening text. This practice helps maintain a clear and organized structure in the document, making it easier for readers to follow.

File: docs/user-guide/nemoaligner.rsts [hunk 14]
```diff
@@ -13,23 +13,48 @@ The NeMo-Aligner toolkit is built using the `NeMo Toolkit <https://github.com/NV
...
 
     Please use the latest tag in the form yy.mm.(patch).
 
-.. note::
+.. important::
    - Some of the subsequent tutorials require accessing gated Hugging Face models. For details on how to access these models, refer to `this document <https://docs.nvidia.com/nemo-framework/user-guide/latest/best-practices.html#working-with-hugging-face-models>`__.
    - If you run into any problems, refer to NeMo's `Known Issues page <https://docs.nvidia.com/nemo-framework/user-guide/latest/knownissues.html>`__. The page enumerates known issues and provides suggested workarounds where appropriate.
 
 
+Building NeMo-Aligner Dockerfile
```
Comment (jgerh) [inline docs/user-guide/nemoaligner.rsts:41]: revise title and use an imperative verb for procedural headings (no "ing" gerunds)

Build a NeMo-Aligner Dockerfile

File: docs/user-guide/nemoaligner.rsts [hunk 15]
```diff
@@ -1,8 +1,8 @@
 
-.. _model-aligner-intro:
+.. _nemo-aligner-intro:
 
-Model Alignment
-!!!!!!!!!!!!!!!
+NeMo-Aligner
```
Comment (jgerh) [inline docs/user-guide/nemoaligner.rsts:4]: suggest title change to be consistent with other titles [Model Alignment by Rejection Sampling, Model Alignment by Supervised Fine-Tuning (SFT), Model Alignment by SteerLM Method]

Model Alignment by NeMo-Aligner or Model Alignment using the NeMo-Aligner Toolkit

**Review by jgerh — COMMENTED** [review]
Completed the technical edit of selected files. Minor copyedits were provided. Please review the comments, make the changes in the files, and resolve the open conversations.

File: docs/user-guide/cai.rst (hunk 1 above)
Comment (terrykong) [reply docs/user-guide/cai.rst:20]: good catch. I've updated it to just say the figure below and the hyperlink is just to the anchor for the figure below.

File: docs/user-guide/nemoaligner.rsts [hunk 16]
```diff
@@ -13,23 +13,48 @@ The NeMo-Aligner toolkit is built using the `NeMo Toolkit <https://github.com/NV
...
+        # Replace with branch to build from
+        ALIGNER_COMMIT=main
+        TARGET_IMAGE=aligner-custom
+
+        docker buildx build \
+            -t $TARGET_IMAGE \
+            --build-arg=ALIGNER_COMMIT=$ALIGNER_COMMIT \
+            .
+
+        # Run the image using the above command in "Getting Started" and swap out "nvcr.io/nvidia/nemo:24.09" with "aligner-custom"
```
[the remark this answers is not in the document]
Comment (terrykong) [reply docs/user-guide/nemoaligner.rsts]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/41c0fd021e313684ea5007ed6948f057e18a2e08

File: docs/user-guide/nemoaligner.rsts (hunk 14 above)
Comment (terrykong) [reply docs/user-guide/nemoaligner.rsts]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/41c0fd021e313684ea5007ed6948f057e18a2e08

File: docs/user-guide/cai.rst (hunk 11 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst [hunk 17]
```diff
@@ -286,7 +283,7 @@ Run the following command to train the RM:
 The trained RM checkpoint will be saved to output dir given by ``exp_manager.explicit_log_dir``.
 
 Step 6: Fine-tune Mistral-7B-SL-CAI with PPO and the RM to train a Mistral-7B-RL-CAI model
```
[the remark this answers is not in the document]
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 10 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 9 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 8 above)
Comment (terrykong) [reply docs/user-guide/cai.rst:112]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst [hunk 18]
```diff
@@ -41,7 +43,7 @@ Simplicity and transparency: encoding the training goals in a simple list of nat
 Reducing iteration time: obviating the need to collect new human feedback labels when altering the objective or testing different behaviors.
 
 Train a CAI model
-#####################
+#################
 
 This section is a step-by-step tutorial that walks you through how to run a full CAI pipeline with a ``Mistral-7B`` LLM model. It includes the following:
```
[the remark this answers is not in the document]
Comment (terrykong) [reply docs/user-guide/cai.rst:46]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 7 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 6 above)
Comment (terrykong) [reply docs/user-guide/cai.rst:21]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 1 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 5 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 4 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 3 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11

File: docs/user-guide/cai.rst (hunk 2 above)
Comment (terrykong) [reply docs/user-guide/cai.rst]: https://github.com/NVIDIA/NeMo-Aligner/pull/411/commits/9432333a44ad01d9534ed89263a37000b2aaca11
